common/src/main/java/com/microsoft/hydralab/performance/PerformanceTestManagementService.java

```diff
@@ -95,19 +110,140 @@ private ITestRun getTestRun() {
+        List<InspectionStrategy> strategyList = testLifeCycleStrategyMap.get(getTestRun().getId());
+        if (strategyList == null) return;
+
+        for (InspectionStrategy inspectionStrategy : strategyList) {
+            if (inspectionStrategy == null) continue;
+            PerformanceInspection inspection = inspectionStrategy.inspection;
+            if (inspectionStrategy.when == null) continue;
+
+            if (whenType == InspectionStrategy.WhenType.TEST_RUN_STARTED) {
+                // initialize inspector
+                PerformanceInspection initialInspection = new PerformanceInspection(
+                        InspectionStrategy.WhenType.TEST_RUN_STARTED.name(),
+                        inspection.inspectorType, inspection.appId, inspection.deviceIdentifier, true);
+                inspect(initialInspection);
+            } else if (inspectionStrategy.when.contains(whenType)) {
+                PerformanceInspection lifeCycleInspection = new PerformanceInspection(
+                        whenType.name() + "-" + description,
+                        inspection.inspectorType, inspection.appId, inspection.deviceIdentifier, false);
+                inspect(lifeCycleInspection);
+            }
+        }
+    }
+
+    /**
+     * For giving inspection return the inspection with device id that related to test run
+     */
+    private PerformanceInspection getDevicePerformanceInspection(PerformanceInspection inspection) {
+        if (getTestRun() instanceof TestRun) {
+            TestRun testRun = (TestRun) getTestRun();
+            return new PerformanceInspection(inspection.description, inspection.inspectorType, inspection.appId,
+                    // For windows inspector, the deviceIdentifier is useless
+                    testRun.getDeviceSerialNumber(), inspection.isReset);
+        }
+        return inspection;
+    }
+
+    private List<PerformanceTestResult> parseForTestRun(ITestRun testRun) {
+        Map<String, PerformanceTestResult> testResultMap = testRunPerfResultMap.get(testRun.getId());
+        if (testResultMap == null) return null;
+
+        List<PerformanceTestResult> resultList = new ArrayList<>();
+        for (PerformanceTestResult performanceTestResult : testResultMap.values()) {
+            PerformanceResultParser parser = getParserByType(performanceTestResult.parserType);
+            Assert.notNull(parser, "Found no matched result parser: " + performanceTestResult.parserType);
+            resultList.add(parser.parse(performanceTestResult));
+        }
+        return resultList;
+    }
+
+    private void savePerfTestResults(List<PerformanceTestResult> resultList) {
+        //TODO save results to DB
+        if (resultList != null && !resultList.isEmpty()) {
+            FileUtil.writeToFile(resultList.toString(),
+                    new File(getTestRun().getResultFolder(), "performance") + File.separator + "PerformanceReport.txt");
```

Comment:
instead of a raw txt file, could it be better if it's a json file or XML file.

Reply:
Good idea. Revised